geomag_imp/secs.py

```diff
@@ -518,146 +522,141 @@ def make_T(self, lat_lon_r):
-                                        x30[n_el] +
-                                    (x3[idx_s] / x30[n_el])**2) +
-                            np.cos(theta_dot[idx_s])))
-
-         # ...then calculate for r>r0
-         idx_f = x3 > x30[n_el]
-         # r-component
-         TX1_dot[idx_f] = (fact * x30[n_el]/x3[idx_f]**2 *
-                           (1 / np.sqrt(1 - 2 * x30[n_el] *
-                                            np.cos(theta_dot[idx_f]) /
-                                            x3[idx_f] +
-                                        (x30[n_el] / x3[idx_f])**2) -
-                            1))
-         # poloidal component
-         TX2_dot[idx_f] = (-fact * 1 / (x3[idx_f] * np.sin(theta_dot[idx_f])) *
-                           ((x3[idx_f] - x30[n_el] * np.cos(theta_dot[idx_f])) /
-                            np.sqrt(x3[idx_f]**2 - 2 * x3[idx_f] * x30[n_el] *
-                                                   np.cos(theta_dot[idx_f]) +
-                                    x30[n_el]**2) -
-                            1))
-
-         # no toroidal component
-         TX3_dot = theta_dot * 0
-
-         # singularities at the pole, TX2_dot is set to zero when theta_dot=0
-         kk = theta_dot==0
-         TX2_dot[kk] = 0
-
-         # transformation of the magnetic field to Cartesian components
-         TX_dot = (np.cos(phi_dot) * np.sin(theta_dot) * TX1_dot +
-                   np.cos(theta_dot) * np.cos(phi_dot) * TX2_dot)
-         TY_dot = (np.sin(phi_dot) * np.sin(theta_dot) * TX1_dot +
-                   np.cos(theta_dot) * np.sin(phi_dot) * TX2_dot)
-         TZ_dot =  np.cos(theta_dot) * TX1_dot - np.sin(theta_dot) * TX2_dot
-
-         # transformation to the cartesian system having pole at theta=0, phi=0
-         TX = (np.cos(x20[n_el]) * TX_dot -
-               np.cos(x10[n_el]) * np.sin(x20[n_el]) * TY_dot -
-               np.sin(x10[n_el]) * np.sin(x20[n_el]) * TZ_dot)
-         TY = (np.sin(x20[n_el]) * TX_dot +
-               np.cos(x10[n_el]) * np.cos(x20[n_el]) * TY_dot +
-               np.sin(x10[n_el]) * np.cos(x20[n_el]) * TZ_dot)
-         TZ = -np.sin(x10[n_el]) * TY_dot + np.cos(x10[n_el]) * TZ_dot
-
-         # transformation back to spherical coordinates
-         TX3 = (np.sin(x1) * np.cos(x2) * TX +
-                np.sin(x1) * np.sin(x2) * TY +
-                np.cos(x1) * TZ) # r-component
-         TX1 = (np.cos(x1) * np.cos(x2) * TX +
-                np.cos(x1) * np.sin(x2) * TY -
-                np.sin(x1) * TZ) # poloidal component
-         TX2 = -np.sin(x2) * TX + np.cos(x2) * TY # toroidal component
-
-         # construct output array
-         # (this differs from Pulkkinen's code, and is more like the literature)
-         T_out[0::3, n_el] = TX1
-         T_out[1::3, n_el] = TX2
-         T_out[2::3, n_el] = TX3
-
-      return T_out
+      nobs = len(lat_lon_r)
+      nsec = len(self.secs.lat_lon_r)
```

Comment:
In the previous code you had `self.secs.n_secs`, I assume these are equivalent, so depending on where you want the consistency you could change this over to that.